src/Sylius/Component/Taxonomy/Factory/TaxonomyFactory.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Taxonomy\Factory;
+
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Taxonomy\Model\TaxonomyInterface;
+use Sylius\Component\Translation\Factory\TranslatableFactory;
+use Sylius\Component\Translation\Provider\LocaleProviderInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class TaxonomyFactory extends TranslatableFactory implements FactoryInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    private $taxonFactory;
+
+    /**
+     * @var string
+     */
+    private $className;
+
+    /**
+     * @param FactoryInterface $taxonFactory
+     */
+    public function __construct(
+        FactoryInterface $translatableResourceFactory,
+        LocaleProviderInterface $localeProvider,
+        FactoryInterface $taxonFactory,
+        $className
+    )
+    {
+        $this->taxonFactory = $taxonFactory;
+        $this->className = $className;
+
+        parent::__construct($translatableResourceFactory, $localeProvider);
+    }
+
+    /**
+     * @return TaxonomyInterface
+     */
+    public function createNew()
+    {
+        $taxon = $this->taxonFactory->createNew();
+
+        $taxonomy = new $this->className($taxon);
+        $taxonomy->setCurrentLocale('en_US');
```

Comment:
You should take these values from LocaleProvider, that's the whole point of injecting it. :)